js/index.js

```diff
@@ -210,7 +217,57 @@ function DataType({covidData, selectedRegions, dataType, checked, onTypeChange})
   )
 }
 
-function CovidChart({covidData, regions, selectedTypes, useLogScale}) {
+function ChartTypeDropDown({ setChartType }) {
+  const chartTypes = ['line', 'bar', 'radar'];
+  const [currChartType, setCurrChartType] = r.useState('Select Chart Type');
+  const onChange = (evt) => {
+    setCurrChartType(evt.target.text)
+    setChartType(evt.target.text)
+  }
+  const dropdownValues = Object.values(chartTypes).map(chartType => {
+    return e('a', { key: chartType, className: 'dropdown-item', value: chartType, onClick: onChange}, chartType)
+  });
+  return e('div', { className: 'btn-group dropleft' },
+    e('button', { 'data-toggle': "dropdown", className: 'btn btn-secondary btn-sm dropdown-toggle' }, currChartType),
+    e('div', { className: 'dropdown-menu' }, dropdownValues)
+    );
+}
+
+function DateRangeSelector({ setStartDate, setEndDate}) {
```

Comment:
What do you think in split this index.js file? I think functions like this can be easily split into your own file. Maybe we can write this function on `src/selectors/date-range-selector.js` and import in this file with `import DateRangeSelector from './src/selectors/date-range-selector'`

Maybe apply this same solution to another function like `ChartTypeDropDown`

Reply:
Yes, we can split the code. But the problem here is that this is a project with just js files and a single index.html file. And No Webserver is required to run this code locally. To use the ES6 import structure properly and easily we have to setup this as a npm project and with package.json and additional tools like Babel, Webpack we can achieve the splitting thing easily.

Without these, we have to use `<script type='module' >` which i tried and getting reference error.
Also See: https://stackoverflow.com/a/950146/9434520

Reply:
I see, but we lost a lot of readability in this `index.js` file, but if works for now, go on :ship:

Reply:
Yup, i saw the code practises used in the file like the usage of Bootstrap and Hooks. And followed the same. So, i think for now its good, we can resolve this conversation.
PS: This is my first time working with Bootstrap and Hooks :see_no_evil: